{
  "data": [
    {
      "_id": 1,
      "x": 11
    }
  ],
  "minServerVersion": "3.6",
  "tests": [
    {
      "description": "First insertOne is never committed",
      "failPoint": {
        "mode": {
          "times": 2
        },
        "data": {
          "failBeforeCommitExceptionCode": 1
        }
      },
      "operation": {
        "name": "bulkWrite",
        "arguments": {
          "requests": [
            {
              "name": "insertOne",
              "arguments": {
                "document": {
                  "_id": 2,
                  "x": 22
                }
              }
            },
            {
              "name": "updateOne",
              "arguments": {
                "filter": {
                  "_id": 2
                },
                "update": {
                  "$inc": {
                    "x": 1
                  }
                }
              }
            },
            {
              "name": "deleteOne",
              "arguments": {
                "filter": {
                  "_id": 1
                }
              }
            }
          ],
          "options": {
            "ordered": true
          }
        }
      },
      "outcome": {
        "error": true,
        "collection": {
          "data": [
            {
              "_id": 1,
              "x": 11
            }
          ]
        }
      }
    },
    {
      "description": "Second updateOne is never committed",
      "failPoint": {
        "mode": {
          "skip": 1
        },
        "data": {
          "failBeforeCommitExceptionCode": 1
        }
      },
      "operation": {
        "name": "bulkWrite",
        "arguments": {
          "requests": [
            {
              "name": "insertOne",
              "arguments": {
                "document": {
                  "_id": 2,
                  "x": 22
                }
              }
            },
            {
              "name": "updateOne",
              "arguments": {
                "filter": {
                  "_id": 2
                },
                "update": {
                  "$inc": {
                    "x": 1
                  }
                }
              }
            },
            {
              "name": "deleteOne",
              "arguments": {
                "filter": {
                  "_id": 1
                }
              }
            }
          ],
          "options": {
            "ordered": true
          }
        }
      },
      "outcome": {
        "error": true,
        "collection": {
          "data": [
            {
              "_id": 1,
              "x": 11
            },
            {
              "_id": 2,
              "x": 22
            }
          ]
        }
      }
    },
    {
      "description": "Third updateOne is never committed",
      "failPoint": {
        "mode": {
          "skip": 2
        },
        "data": {
          "failBeforeCommitExceptionCode": 1
        }
      },
      "operation": {
        "name": "bulkWrite",
        "arguments": {
          "requests": [
            {
              "name": "insertOne",
              "arguments": {
                "document": {
                  "_id": 2,
                  "x": 22
                }
              }
            },
            {
              "name": "updateOne",
              "arguments": {
                "filter": {
                  "_id": 1
                },
                "update": {
                  "$inc": {
                    "x": 1
                  }
                }
              }
            },
            {
              "name": "updateOne",
              "arguments": {
                "filter": {
                  "_id": 2
                },
                "update": {
                  "$inc": {
                    "x": 1
                  }
                }
              }
            }
          ],
          "options": {
            "ordered": true
          }
        }
      },
      "outcome": {
        "error": true,
        "collection": {
          "data": [
            {
              "_id": 1,
              "x": 12
            },
            {
              "_id": 2,
              "x": 22
            }
          ]
        }
      }
    },
    {
      "description": "InsertMany fails after multiple network errors",
      "failPoint": {
        "mode": {
          "skip": 0
        },
        "data": {
          "failBeforeCommitExceptionCode": 1
        }
      },
      "operation": {
        "name": "insertMany",
        "arguments": {
          "documents": [
            {
              "_id": 2,
              "x": 22
            },
            {
              "_id": 3,
              "x": 33
            },
            {
              "_id": 4,
              "x": 44
            }
          ],
          "options": {
            "ordered": true
          }
        }
      },
      "outcome": {
        "error": true,
        "collection": {
          "data": [
            {
              "_id": 1,
              "x": 11
            }
          ]
        }
      }
    }
  ]
}
